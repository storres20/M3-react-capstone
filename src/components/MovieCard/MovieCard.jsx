import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './MovieCard.module.css';

export default function MovieCard({ movie }) {
  const imageUrl = `https://image.tmdb.org/t/p/w300${movie.poster_path}`;
  return (
    <li className={styles.movieCard}>
      <Link to={`/movie/${movie.id}`}>
        <img
          width={230}
          height={345}
          className={styles.movieImage}
          src={imageUrl}
          alt={movie.title}
        />
        <div>{movie.title}</div>
      </Link>
    </li>
  );
}

MovieCard.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    poster_path: PropTypes.string,
  }).isRequired,
};
