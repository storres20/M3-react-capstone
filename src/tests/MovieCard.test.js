import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import MovieCard from '../components/MovieCard/MovieCard';

it('renders correctly', () => {
  const movie = {
    id: 399566,
    title: 'Godzilla vs. Kong',
    poster_path: '/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg',
  };

  const tree = renderer
    .create(
      <BrowserRouter>
        <MovieCard movie={movie} />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

MovieCard.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    poster_path: PropTypes.string,
  }),
};
