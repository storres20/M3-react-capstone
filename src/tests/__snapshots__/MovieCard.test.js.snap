// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="movieCard"
>
  <a
    href="/movie/399566"
    onClick={[Function]}
  >
    <img
      alt="Godzilla vs. Kong"
      className="movieImage"
      height={345}
      src="https://image.tmdb.org/t/p/w300/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
      width={230}
    />
    <div>
      Godzilla vs. Kong
    </div>
  </a>
</li>
`;
